### ISSUE TITLE
Publish version 1.2 to App Store

### ISSUE THREAD

**Issue by aaronpk** [aaronpk/Overland-iOS#41]

**Comment by groupnet** [comment]
Do you already have an idea when the next release will be published in the App Store?
I'm really looking forward to the [url scheme for overland](https://github.com/aaronpk/Overland-iOS/pull/68) being available again.

**Comment by aaronpk** [comment]
I was hoping to get a few more features in, but at this point there's a pretty good set of minor changes and bugfixes that I should probably just push a release anyway. I probably won't have time to do that until the end of the month though.

**Comment by groupnet** [comment]
Sounds great - thanks for the quick reply and providing such a fabulous tool!

**Comment by groupnet** [comment]
Any news on when the next release will be published in the App Store?

**Comment by groupnet** [comment]
Any news?

**Comment by fivestones** [comment]
Would love to see this get pushed to the App Store! 👍

**Comment by aaronpk** [comment]
I just finished a marathon coding session to close out all of the new features I wanted to get in to this release!

Before I push this out to the app store, I'd love to get a few beta testers to provide feedback! If you'd like to test it out, sign up here and I'll send you a TestFlight invite!

https://forms.gle/8HvoPsMFgTwN4uup8

**Comment by momonala** [comment]
Hey @aaronpk, already enjoying this new update! Im testing it on TestFlight. Is this the best place for feedback?

The new time and distance filters are great, and the dark mode support. I have a question about the Configure Wifi Zone thing. The first time I clicked it, it said it registered my home wifi as the zone and set the long/lat. However, Im at work now and wanted to set another wifi zone. I clicked configure again but there wasnt much of a UI response and the old wifi name was automatically populated in the form. So its unclear so far if I can set multiple wifi zones, or where my current zone is configured. Anything Im missing here to make it more clear? Thanks!

**Comment by aaronpk** [comment]
Yeah the wifi zone thing is a bit of a hack, and I'm definitely not good at UI stuff on iOS. 

Right now there can be only one wifi zone configured.

The way it works is the first time you click it, it will prefill the location and wifi name and you'll see "Save" and "Cancel". If you already have a wifi zone configured, the buttons will say "Save" and "Clear".

This definitely needs some improvement, and I'd also like to be able to set more than one. That is all going to take a significant amount of coding though, and in particular the kind of iOS coding that I am not good at!

**Comment by aaronpk** [comment]
Thanks! I'm going to move that thread to a new issue to discuss there

**Comment by uqe** [comment]
@aaronpk, sent my request to Google Forms, is it still possible to get access to the beta?

**Comment by aaronpk** [comment]
Thanks! I just added you on TestFlight

**Comment by aaronpk** [comment]
FYI I just published a new build to TestFlight!

The most significant change is an internal change that should allow the app to track better in the background on the most recent iOS versions.

There are also a number of new settings for you!

* Adds a setting to toggle the system background location indicator
* Adds URL template to send location data in the URL instead of POST body (See GitHub README for docs)
* Adds a setting to log only the latest location instead of full history

Please continue to test and let me know how it goes! I would love to push this out for real soon!

**Comment by iamchrishckns** [comment]
Hey @aaronpk,

love the new features. 

One thing I noticed is that switching the "Logging Mode" to "Only Latest" clears the already queued locations without a warning. I wondered why my queue was empty after trying it out. This time it doesn't bother me but when using Overland on the go I'm a bit worried to clear the queue by accident.

Would it be possible to display a confirmation dialog when the queue has locations queued instead of just clearing it? A bit like you already mentioned in #78 for the "Purge Queue" feature?

Best regards

**Comment by aaronpk** [comment]
oh that's a great point! Well I was hoping to not have to build the confirmation dialog this time around, but sounds like I will need to do it anyway because I inadvertently added a purge queue button :joy:

**Comment by eiiot** [comment]
Also filled out the TestFlight form!

**Comment by aaronpk** [comment]
@eiiot thanks for the nudge! I just sent your invite

**Comment by oliverpool** [comment]
This version sounds amazing! Is there any release blocker? Or do you need more testers?

**Comment by aaronpk** [comment]
I'm getting a crash occasionally when ending a trip, but re-launching the app and ending the trip works fine, and I don't seem to be losing any data over it. Other than that things seem to be going well. The last thing I need to do is some more testing on the new "tip jar" feature, but it's about ready to go!

**Comment by jasonsilberman** [comment]
Hey @aaronpk, I submitted the form to be added to the TestFlight. I've been building out a Cloudflare backend for the service and I would love to try out the new version!

**Comment by oliverpool** [comment]
@jasonsilberman I also plan to develop a backend (eventually on Cloudflare)- for team tracking.

do you plan to share your code?

**Comment by jasonsilberman** [comment]
> @jasonsilberman I also plan to develop a backend (eventually on Cloudflare)- for team tracking.
> 
> 
> 
> do you plan to share your code?

@oliverpool yes, i do plan on opening sourcing the code. however, it's still in the quite early stages right now!

**Comment by aaronpk** [comment]
@jasonsilberman sorry about that, I am not getting notifications from the google sheet it seems! I just sent you (and a few others) the invite.

**Comment by oleg131** [comment]
Have been testing it for a few days and this is what I have to report:

Even when continuous tracking is off, and only visit tracking is on, the battery usage is very high. iOS reports is as being 25-33%. For comparison, my other 2 apps that use background location tracking - Life360 and Life Cycle - are reported to use less than 1%.

My sent interval is set to 30m, other settings are on the screenshot. Am I using it correctly?

![IMG_9721](https://github.com/aaronpk/Overland-iOS/assets/10161825/888f2a6e-c758-44f3-a862-ff326af42c7e)

**Comment by aaronpk** [comment]
It's probably the "Desired Accuracy: Best" that's causing that. Have you tried setting that to a less accurate value? 

I'm assuming you don't get exact trails recorded in the other apps? IIRC those are doing mostly enter/exit region tracking to tell you how long you spent at certain places.

**Comment by oleg131** [comment]
Looks like the least battery intenstive location tracking is using `startMonitoringSignificantLocationChanges` which is what the other apps are using I assume.

If I turn off Visit Tracking and set Continuous Tracking Mode to Significant, would that achieve this?

The top Tracking Enabled toggle should still be set to On, right?

**Comment by aaronpk** [comment]
Yes, Significant Location will be the most battery efficient, but will result in the least amount of data collected. This is basically always a tradeoff between battery and accuracy. Visit tracking should be at a similar low power usage similar to Significant, so you can leave that on.

The top "Tracking Enabled" is a global on/off for the entire app, if you turn that off then the app will turn off all tracking and stop logging anything.

This app originally started as a testing tool to see the effect of all these iOS settings, and evolved into how it's being used today. I'm starting to think I need to move all these settings to an "Advanced" page, and make essentially some presets based on common uses. But I think that will have to wait for the next version because I'd really like to get this out the door.

**Comment by oleg131** [comment]
After few days of testing:

I've disabled Visit Tracking and set Continuous Tracking Mode to Significant, and battery usage went down to "low". I'm using test endpoint where I'm just printing the payload and it seems to work fine so far.

**Comment by acaranta** [comment]
Just filled the testflight form, if it's possible to be included and testing :)
(thx for your work btw :) )

**Comment by rexxars** [comment]
Would also love to be included! Testflight form filled.

**Comment by momonala** [comment]
Hey @aaronpk , my TestFlight app has expired! Any chance you can renew it? Or is the app close to being updated on the AppStore? I see its still 1.2

**Comment by aaronpk** [comment]
Oh shoot, yeah I can put out another TestFlight version today! 

I'm pretty happy with how this version has been working so I'm almost ready to release it to the App Store. The one thing left I've been working on adding is an in-app purchase tip jar, which has turned out to be somewhat complicated, so I need to decide to either finish that or give up on it.

**Comment by aaronpk** [comment]
New beta version is out! I added two new features as well:

* Home Assistant support
* More properties in location data: course, course accuracy, speed accuracy

**Comment by radh21301** [comment]
Just filled out the google form. Hopefully you are still giving people access to the beta releases. Really excited to use the app!

**Comment by aaronpk** [comment]
I just submitted this to Apple for review! Cross your fingers! 🤞

**Comment by aaronpk** [comment]
I had to add a link to apple's EULA on the in-app purchase page. Resubmitting the app today!

**Comment by aaronpk** [comment]
Now they want me to add a "restore purchases" button to restore the ... tips? The in-app purchases are only tips to the developer (which is allowed), so the purchase doesn't unlock any functionality. So what exactly is the "restore" button going to do...?

**Comment by aaronpk** [comment]
I added a "restore purchases" button which restores the purchases which does... nothing.

> Thank you for your resubmission. Upon further review, we identified additional issues that need your attention. See below for more information.
> We have started the review of your app, but we are not able to continue because we cannot locate the in-app purchases within your app at this time.

🤦‍♂️

**Comment by aaronpk** [comment]
They approved it!

**Comment by aaronpk** [comment]
Version 1.3 is released and making its way to the App Store! Thanks everyone for your patience!
